### ISSUE TITLE
pdfn arg rewrite does not respect bindings

### ISSUE THREAD

**Issue by selfsame** [selfsame/pdf#21]
```clj
(defpdfn ^:inline goat)

(pdfn goat [d c]
  (let [[a b c d] [:a :b :c :d]]
    [a b c d]))

(inspect goat)

(goat 1 2)
```

**Comment by selfsame** [comment]
```clj
(def goat 
  (fn ([a b] 
    (let [[a b b a] [:a :b :c :d]] 
      [a b b a]))))
```

**Comment by selfsame** [comment]
problem here is collisions with pdfn's generated `abcd..` argument symbols, using gensyms would avoid this

**Comment by selfsame** [comment]
fixed with 7579ad9ea12b9cc4ab568336f0e1db4b54fda8b4
